hydra-node/test/Hydra/NetworkSpec.hs

```diff
@@ -52,8 +54,14 @@ spec = do
     prop "can roundtrip CBOR encoding/decoding of Hydra Message" $ prop_canRoundtripCBOREncoding @(Message SimpleTx)
     roundtripAndGoldenSpecs (Proxy @(Message SimpleTx))
 
+  describe "Message Persistence" $ do
+    it "throws ParameterMismatch when configuring given number of acks does not match number of parties" $ do
+      withTempDir "persistence" $ \dir -> do
+        writeFile (acksFile dir) "[0,0,0]"
+        configureMessagePersistence @_ @Int nullTracer dir 4 `shouldThrow` (const True :: Selector ParameterMismatch)
```

Comment:
Should: avoid writing the internal representation here.

Instead, we can express ourselves by using the interfaces provided of the "system under test", here: `configureMessagePersistence`

```suggestion
        MesagePersistence{saveAcks} <- configureMessagePersistence @_ @Int nullTracer dir 3
        saveAcks (fromList [0, 0, 0])
        configureMessagePersistence @_ @Int nullTracer dir 4 `shouldThrow` (const True :: Selector ParameterMismatch)
```